src/Microsoft.Data.Analysis/DataFrameRowCollection.cs

```diff
@@ -0,0 +1,130 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Data.Analysis
+{
+
+    /// <summary>
+    /// Represents the rows of a <see cref="DataFrame"/>
+    /// </summary>
+    public class DataFrameRowCollection : IEnumerable<DataFrameRow>
+    {
+        private long _currentIndex = 0;
```

Comment:
Nit:
```suggestion
        private long _currentIndex;
```
Zero-init is done by default. Though I have no strong opinion on this.

Reply:
Anyway this doesn't need to be a field (see comment below).